cmds/rm/rm_test.go

```diff
@@ -7,114 +7,246 @@
 package main
 
 import (
-	"fmt"
+	"bufio"
 	"io/ioutil"
 	"os"
-	"path/filepath"
+	"path"
 	"testing"
 )
 
-type makeit struct {
-	n string      // name
-	m os.FileMode // mode
-	s string      // for symlinks or content
+type file struct {
+	name   string
+	delete bool
 }
 
-var tests = []makeit{
-	{
-		n: "hi1.txt",
-		m: 0666,
-		s: "",
-	},
-	{
-		n: "hi2.txt",
-		m: 0777,
-		s: "",
-	},
-	{
-		n: "go.txt",
-		m: 0555,
-		s: "",
-	},
+type rmTestCase struct {
+	name  string
+	files []file
+	i     bool
+	r     bool
+	f     bool
+	err   func(error) bool
+	stdin *fakeStdin
 }
 
-// initial files and folders (for testing)
-func setup() (string, error) {
-	fmt.Println(":: Creating simulating data...")
-	d, err := ioutil.TempDir(os.TempDir(), "hi.dir")
-	if err != nil {
-		return "", err
-	}
-
-	tmpdir := filepath.Join(d, "hi.sub.dir")
-	if err := os.Mkdir(tmpdir, 0777); err != nil {
-		return "", err
+var (
+	no       = newFakeStdin("no")
+	fbody    = []byte("Go is cool!")
+	tmpFiles = []struct {
```

Comment:
Move these global vars inside the test? Doesn't look like this is touched outside that function.